{"ast":null,"code":"export default async function get() {\n  const url = \"https://cubioinfo.com/aptitude-test/api/testitems/1\";\n\n  try {\n    const response = await fetch(url);\n\n    if (response.ok) {\n      return await response.clone().json().catch(() => response.text()); // return await response.json();\n    }\n\n    if (response.status <= 501) {// eslint-disable-next-line prefer-promise-reject-errors\n    }\n  } catch (error) {\n    if (error.response) {\n      /*\n       * The request was made and the server responded with a\n       * status code that falls out of the range of 2xx\n       */\n      return error.response.status;\n    } else if (error.request) {\n      /*\n       * The request was made but no response was received, `error.request`\n       * is an instance of XMLHttpRequest in the browser and an instance\n       * of http.ClientRequest in Node.js\n       */\n      console.log(error.request);\n    } else {\n      // Something happened in setting up the request and triggered an Error\n      console.log(\"Error\", error.message);\n    }\n\n    console.log(error.config);\n  }\n}","map":{"version":3,"sources":["/Users/apple/Documents/GitHub/OCSCE/src/service/get.js"],"names":["get","url","response","fetch","ok","clone","json","catch","text","status","error","request","console","log","message","config"],"mappings":"AAAA,eAAe,eAAeA,GAAf,GAAqB;AAClC,QAAMC,GAAG,GAAG,qDAAZ;;AACA,MAAI;AACF,UAAMC,QAAQ,GAAG,MAAMC,KAAK,CAACF,GAAD,CAA5B;;AACA,QAAIC,QAAQ,CAACE,EAAb,EAAiB;AACf,aAAO,MAAMF,QAAQ,CAClBG,KADU,GAEVC,IAFU,GAGVC,KAHU,CAGJ,MAAML,QAAQ,CAACM,IAAT,EAHF,CAAb,CADe,CAKf;AACD;;AACD,QAAIN,QAAQ,CAACO,MAAT,IAAmB,GAAvB,EAA4B,CAC1B;AACD;AACF,GAZD,CAYE,OAAOC,KAAP,EAAc;AACd,QAAIA,KAAK,CAACR,QAAV,EAAoB;AAClB;;;;AAIA,aAAOQ,KAAK,CAACR,QAAN,CAAeO,MAAtB;AACD,KAND,MAMO,IAAIC,KAAK,CAACC,OAAV,EAAmB;AACxB;;;;;AAKAC,MAAAA,OAAO,CAACC,GAAR,CAAYH,KAAK,CAACC,OAAlB;AACD,KAPM,MAOA;AACL;AACAC,MAAAA,OAAO,CAACC,GAAR,CAAY,OAAZ,EAAqBH,KAAK,CAACI,OAA3B;AACD;;AACDF,IAAAA,OAAO,CAACC,GAAR,CAAYH,KAAK,CAACK,MAAlB;AACD;AACF","sourcesContent":["export default async function get() {\n  const url = \"https://cubioinfo.com/aptitude-test/api/testitems/1\";\n  try {\n    const response = await fetch(url);\n    if (response.ok) {\n      return await response\n        .clone()\n        .json()\n        .catch(() => response.text());\n      // return await response.json();\n    }\n    if (response.status <= 501) {\n      // eslint-disable-next-line prefer-promise-reject-errors\n    }\n  } catch (error) {\n    if (error.response) {\n      /*\n       * The request was made and the server responded with a\n       * status code that falls out of the range of 2xx\n       */\n      return error.response.status;\n    } else if (error.request) {\n      /*\n       * The request was made but no response was received, `error.request`\n       * is an instance of XMLHttpRequest in the browser and an instance\n       * of http.ClientRequest in Node.js\n       */\n      console.log(error.request);\n    } else {\n      // Something happened in setting up the request and triggered an Error\n      console.log(\"Error\", error.message);\n    }\n    console.log(error.config);\n  }\n}\n"]},"metadata":{},"sourceType":"module"}